Library/Homebrew/utils.rb

```diff
@@ -291,6 +291,18 @@ def paths
   end.uniq.compact
 end
 
+# return the shell profile file based on users' preference shell
+def shell_profile
+  case ENV["SHELL"]
+  when %r{/bash} then "~/.bash_profile"
+  when %r{/zsh} then "~/.zshrc"
+  when %r{/t?csh} then "~/.cshrc"
+  when %r{/ksh} then "~/.kshrc"
+  when %r{/sh} then "~/.bash_profile"
+  else "~/.profile"
```

Comment:
Maybe should default to `.bash_profile` still? Does `~/.profile` get read by default on Bash/ZSH?